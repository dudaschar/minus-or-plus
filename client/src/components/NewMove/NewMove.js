import React from 'react';
import PropTypes from 'prop-types';

import { CatIcon } from 'src/icons/cat';
import { EtIcon } from 'src/icons/et';

import { Move, MoveDetails } from './styles/Move';
import { Container, MoveWrapper, IconWrapper } from './styles/Container';

function NewMove({ content, currentPlayer }) {
  const { number, addition, nextNumber, player } = content;

  const isALocalMove = currentPlayer === player ? true : false;
  const isACatPlayer = player === 'player1';
  return (
    <Container isALocalMove={isALocalMove}>
      <IconWrapper>{isACatPlayer ? <CatIcon /> : <EtIcon />}</IconWrapper>
      <MoveWrapper isALocalMove={isALocalMove}>
        <Move>{addition}</Move>
        <MoveDetails>
          [({addition} + {number}) / 3] = {nextNumber}
        </MoveDetails>
        <MoveDetails>{nextNumber}</MoveDetails>
      </MoveWrapper>
    </Container>
  );
}

NewMove.propTypes = {
  content: PropTypes.shape({
    number: PropTypes.number,
    addition: PropTypes.number,
    nextNumber: PropTypes.number,
    player: PropTypes.string,
  }).isRequired,
  currentPlayer: PropTypes.string,
};

export default NewMove;
