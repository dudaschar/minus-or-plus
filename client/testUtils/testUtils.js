import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from '../stylesUtils/theme';

const TestingWithProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

TestingWithProviders.propTypes = {
  children: PropTypes.any.isRequired,
};

const customRender = (ui, options) => (
  render(ui,  { wrapper: TestingWithProviders, ...options })
);

export * from '@testing-library/react';
export { customRender as render };

