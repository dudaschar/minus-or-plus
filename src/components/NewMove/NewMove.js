import React from 'react';
import PropTypes from 'prop-types';

import { CatIcon } from  'src/icons/cat';

import { Move, MoveDetails } from './styles/Move';
import { Container, MoveWrapper, IconWrapper } from './styles/Container';

function NewMove({ content }) {
  const {
    initialNumber,
    addition,
    resultingNumber
  } = content;

  return (
    <Container>
      <IconWrapper>
        <CatIcon />
      </IconWrapper>
      <MoveWrapper>
        <Move>{addition}</Move>
        <MoveDetails>
          [({addition} + {initialNumber}) / 3] = {resultingNumber}
        </MoveDetails>
        <MoveDetails>
          {resultingNumber}
        </MoveDetails>
      </MoveWrapper>
    </Container>
  );
}

NewMove.propTypes = {
  content: PropTypes.shape({
    initialNumber: PropTypes.number,
    addition: PropTypes.number,
    resultingNumber: PropTypes.number,
  }),
};

export default NewMove;
