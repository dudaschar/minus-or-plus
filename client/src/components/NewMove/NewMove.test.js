import React from 'react';
import { render, screen } from 'testUtils/testUtils';

import NewMove from './NewMove';

describe('NewMove Component', () => {
  it('should render properly with player 1', () => {
    const content = {number: 6, addition: 0, nextNumber: 2, player: 'player1'};
    const { asFragment } = render(<NewMove content={content} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render properly with player 2', () => {
    const content = {number: 6, addition: 0, nextNumber: 2, player: 'player2'};
    const { asFragment } = render(<NewMove content={content} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the expression', () => {
    const content = {number: 6, addition: 0, nextNumber: 2, player: 'player1'};
    render(<NewMove content={content} />);

    expect(screen.getByText('[(0 + 6) / 3] = 2')).toBeTrue;
  });
});