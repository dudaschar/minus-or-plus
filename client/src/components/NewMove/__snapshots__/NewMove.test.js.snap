// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMove Component should render properly with player 1 1`] = `
<DocumentFragment>
  <div
    class="Container-coo0jl-0 hSvYis"
  >
    <div
      class="Container__IconWrapper-coo0jl-2 eoYbCv"
    >
      <svg
        height="40px"
        viewBox="0 0 511.999 511.999"
        width="40px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M167.119 91.925L49.156 10.915C37.405 2.74 21.655 12.928 24.292 26.998l34.72 186.242c18.153-53.577 57.501-97.357 108.107-121.315z"
          fill="#ffd164"
        />
        <path
          d="M137.294 71.443a234.154 234.154 0 00-18.941 12.065c-5.479 3.865-12.718 4.113-18.245.317L75.094 66.647c-5.182-3.605-12.128.888-10.965 7.093l7.638 40.971c1.136 6.093-.572 12.299-4.417 17.16-6.747 8.533-12.952 17.513-18.483 26.947l10.145 54.421c18.153-53.577 57.501-97.357 108.107-121.315l-29.825-20.481z"
          fill="#ffc350"
        />
        <path
          d="M344.88 91.925l117.963-81.011c11.751-8.174 27.501 2.014 24.863 16.083L452.988 213.24c-18.154-53.577-57.502-97.357-108.108-121.315z"
          fill="#ffd164"
        />
        <path
          d="M374.705 71.443a234.154 234.154 0 0118.941 12.065c5.479 3.865 12.718 4.113 18.245.317l25.014-17.178c5.182-3.605 12.128.888 10.965 7.093l-7.638 40.971c-1.136 6.093.572 12.299 4.417 17.16 6.747 8.533 12.952 17.513 18.483 26.947l-10.145 54.421c-18.153-53.577-57.501-97.357-108.107-121.315l29.825-20.481z"
          fill="#ffc350"
        />
        <circle
          cx="256"
          cy="280"
          fill="#ffe17d"
          r="224"
        />
        <path
          d="M288.516 471.483c-123.712 0-224-100.288-224-224 0-53.573 18.831-102.734 50.204-141.28C64.255 147.276 32 209.861 32 280c0 123.711 100.288 224 224 224 70.139 0 132.725-32.256 173.797-82.721-38.546 31.372-87.707 50.204-141.281 50.204z"
          fill="#ffd164"
        />
        <path
          d="M256 349.793c-47.782 0-93.392-4.854-135.236-13.629-5.42-1.137-10.091 3.818-8.419 8.981C131.642 404.756 188.602 448 256 448s124.358-43.244 143.655-102.854c1.672-5.164-2.999-10.118-8.419-8.981-41.844 8.774-87.454 13.628-135.236 13.628z"
          fill="#9c6846"
        />
        <path
          d="M391.524 364.915c3.158-6.359 5.92-12.941 8.13-19.77 1.672-5.164-2.999-10.118-8.419-8.981-41.844 8.775-87.454 13.629-135.236 13.629s-93.392-4.854-135.236-13.629c-5.42-1.137-10.091 3.817-8.419 8.981 2.21 6.829 4.972 13.411 8.13 19.77C161.747 373.622 207.617 378.5 256 378.5s94.253-4.878 135.524-13.585z"
          fill="#7d5046"
        />
        <path
          d="M278.919 272H233.08c-6.783 0-10.488 7.911-6.146 13.121l22.92 27.504c3.198 3.838 9.093 3.838 12.292 0l22.92-27.504c4.341-5.21.636-13.121-6.147-13.121z"
          fill="#9c6846"
        />
        <path
          d="M376.007 242a9.974 9.974 0 01-7.805-3.734C361.851 230.359 353.25 226 344 226s-17.852 4.359-24.203 12.266c-3.453 4.305-9.758 5-14.063 1.531-4.305-3.461-4.992-9.758-1.531-14.062C314.289 213.195 328.789 206 344 206c15.211 0 29.711 7.195 39.797 19.734 3.461 4.305 2.773 10.602-1.531 14.062a9.955 9.955 0 01-6.259 2.204zM199.976 242a9.974 9.974 0 01-7.805-3.734C185.82 230.359 177.219 226 167.969 226s-17.852 4.359-24.203 12.266c-3.461 4.305-9.758 5-14.062 1.531-4.305-3.461-4.992-9.758-1.531-14.062C138.258 213.195 152.758 206 167.969 206s29.711 7.195 39.797 19.734c3.461 4.305 2.773 10.602-1.531 14.062a9.955 9.955 0 01-6.259 2.204zM184 280H8c-4.422 0-8-3.578-8-8s3.578-8 8-8h176c4.422 0 8 3.578 8 8s-3.578 8-8 8zM7.992 336c-3.867 0-7.273-2.813-7.891-6.75a7.99 7.99 0 016.648-9.148l152-24c4.344-.711 8.469 2.289 9.148 6.648a7.99 7.99 0 01-6.648 9.148l-152 24c-.421.07-.843.102-1.257.102zM503.999 280H328c-4.422 0-8-3.578-8-8s3.578-8 8-8h176c4.422 0 8 3.578 8 8s-3.579 8-8.001 8zM504.007 336c-.414 0-.836-.031-1.258-.102l-152-24c-4.359-.688-7.344-4.781-6.648-9.148.688-4.359 4.789-7.352 9.148-6.648l152 24c4.359.688 7.344 4.781 6.648 9.148-.616 3.937-4.023 6.75-7.89 6.75z"
          fill="#aa7346"
        />
        <path
          d="M256 349.793c-30.35 0-59.811-1.973-88-5.651v39.543c0 7.127 8.617 10.697 13.657 5.657l10.497-10.498a7.899 7.899 0 016.269-2.289A682.405 682.405 0 00256 379c19.606 0 38.814-.865 57.567-2.453a7.898 7.898 0 016.27 2.29l10.506 10.506c5.04 5.04 13.657 1.47 13.657-5.657v-39.543c-28.189 3.677-57.651 5.65-88 5.65z"
          fill="#fff"
        />
        <path
          d="M256 292c-12.018 0-22.899-4.83-30.84-12.64-.17 1.959.315 4.009 1.775 5.761l22.919 27.503c3.199 3.838 9.093 3.838 12.292 0l22.919-27.503c1.46-1.752 1.945-3.802 1.775-5.761C278.899 287.17 268.017 292 256 292z"
          fill="#7d5046"
        />
      </svg>
    </div>
    <div
      class="Container__MoveWrapper-coo0jl-1 jNIrZL"
    >
      <p
        class="Move-sc-1x0civq-0 feXDtJ"
      >
        0
      </p>
      <p
        class="Move__MoveDetails-sc-1x0civq-1 fglXIg"
      >
        [(0 + 6) / 3] = 2
      </p>
      <p
        class="Move__MoveDetails-sc-1x0civq-1 fglXIg"
      >
        2
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NewMove Component should render properly with player 2 1`] = `
<DocumentFragment>
  <div
    class="Container-coo0jl-0 hSvYis"
  >
    <div
      class="Container__IconWrapper-coo0jl-2 eoYbCv"
    >
      <svg
        height="40px"
        viewBox="0 0 512 512"
        width="40px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.226 231.226C487.226 358.928 356.338 512 256 512S24.774 358.928 24.774 231.226 128.298 0 256 0s231.226 103.523 231.226 231.226z"
          fill="#c9c6ca"
        />
        <path
          d="M256 423.226c-19.46 0-37.282-7.234-48.887-19.847a10.325 10.325 0 01.605-14.589c4.194-3.847 10.734-3.589 14.589.605 7.597 8.258 20.194 13.185 33.694 13.185s26.097-4.927 33.694-13.185c3.855-4.194 10.403-4.452 14.589-.605a10.325 10.325 0 01.605 14.589c-11.607 12.613-29.429 19.847-48.889 19.847z"
          fill="#6e5a64"
        />
        <path
          d="M305.548 495.484c-100.338 0-231.226-161.33-231.226-289.032 0-80.667 41.334-151.65 103.951-193.022-89.459 31.93-153.5 117.368-153.5 217.796C24.774 358.928 155.662 512 256 512c20.459 0 42.186-6.45 63.817-17.665-4.806.685-9.58 1.149-14.269 1.149z"
          fill="#b7b2b8"
        />
        <path
          d="M172.464 337.709c-31.589 4.625-60.946-17.233-65.571-48.822l-7.178-49.026c-4.625-31.589 17.233-60.946 48.822-65.571 31.589-4.625 60.946 17.233 65.571 48.822l7.178 49.026c4.625 31.589-17.233 60.946-48.822 65.571z"
          fill="#6e5a64"
        />
        <path
          d="M194.584 317.778c-31.589 4.625-60.946-17.233-65.571-48.822l-11.653-79.585c-13.292 12.505-20.49 31.063-17.646 50.49l7.178 49.026c4.625 31.589 33.983 53.447 65.571 48.822 17.081-2.501 31.267-12.267 40.009-25.673-5.476 2.796-11.466 4.802-17.888 5.742z"
          fill="#5a4650"
        />
        <path
          d="M339.536 337.709c-31.589-4.625-53.447-33.982-48.822-65.571l7.178-49.026c4.625-31.589 33.982-53.447 65.571-48.822s53.447 33.982 48.822 65.571l-7.178 49.026c-4.625 31.589-33.982 53.448-65.571 48.822z"
          fill="#6e5a64"
        />
        <path
          d="M366.442 324.957c-31.589-4.625-53.447-33.983-48.822-65.571l11.653-79.585c-16.319 8.17-28.535 23.885-31.38 43.312l-7.178 49.026c-4.625 31.589 17.233 60.946 48.822 65.571 17.081 2.501 33.471-2.788 45.69-13.125-6.049 1.107-12.363 1.312-18.785.372z"
          fill="#5a4650"
        />
      </svg>
    </div>
    <div
      class="Container__MoveWrapper-coo0jl-1 jNIrZL"
    >
      <p
        class="Move-sc-1x0civq-0 feXDtJ"
      >
        0
      </p>
      <p
        class="Move__MoveDetails-sc-1x0civq-1 fglXIg"
      >
        [(0 + 6) / 3] = 2
      </p>
      <p
        class="Move__MoveDetails-sc-1x0civq-1 fglXIg"
      >
        2
      </p>
    </div>
  </div>
</DocumentFragment>
`;
